### ISSUE TITLE
[DX] Mark user_filters() and user_build_filter_query() as deprecated

### ISSUE THREAD

**Issue by avpaderno** [backdrop/backdrop-issues#6534]
In Drupal 7, [`user_filters()`](https://api.drupal.org/api/drupal/modules%21user%21user.module/function/user_filters/7.x) is called from [`user_admin()`](https://api.drupal.org/api/drupal/modules%21user%21user.admin.inc/function/user_admin/7.x), the page callback for admin/people, via [`user_filter_form()`](https://api.drupal.org/api/drupal/modules%21user%21user.admin.inc/function/user_filter_form/7.x).

In Backdrop, admin/people is a view. [`user_filters()`](https://docs.backdropcms.org/api/backdrop/core%21modules%21user%21user.module/function/user_filters/1) is not used to build options for a form element that is shown to people with an account on a Backdrop site. It is called only from [`user_build_filter_query()`](https://docs.backdropcms.org/api/backdrop/core%21modules%21user%21user.module/function/user_build_filter_query/1), which is then only called by [`database_test_theme_tablesort()`](https://docs.backdropcms.org/api/backdrop/core%21modules%21simpletest%21tests%21database_test.module/function/database_test_theme_tablesort/1), a test function whose purpose is testing [`theme_tableselect()`](https://docs.backdropcms.org/api/backdrop/core%21includes%21form.inc/function/theme_tableselect/1).

`database_test_theme_tablesort()` should be rewritten to avoid `user_build_filter_query()`. `user_filters()` and `user_build_filter_query()` should be marked deprecated.

**Comment by klonos** [comment]
Thank you @kiamlaluno for providing the background info for these functions and their usage 🙏🏼 ...for some visuals, this is where they were used in D7:
![image](https://github.com/backdrop/backdrop-issues/assets/2423362/c6a21978-76b2-4958-897f-fa94dcb386df)

**Comment by klonos** [comment]
...I initially set the 2.x milestone for this task, but we could add the deprecated warnings in 1.x, as well as rewrite `database_test_theme_tablesort()` to not be calling `user_build_filter_query()`.

**Comment by klonos** [comment]
@kiamlaluno in #6456 you provided a PR that also properly translated the permission strings (backporting the fix suggested in https://www.drupal.org/project/drupal/issues/1956016). We should include that fix here, in case this is being used by custom/contrib code that was not adjusted in Backdrop yet. Besides, we don't want to be translating permission machine names, and if we leave it as it is, it will be included in the strings translators need to translate.

As I said in that other issue, I still don't get why we are doing this in the options array keys:
`t('@module module', array('@module' => $module))`

...we should either remove the call to `t()`, or add a comment explaining why that is done.

**Comment by klonos** [comment]
> As I said in that other issue, I still don't get why we are doing this in the options array keys:
> `t('@module module', array('@module' => $module))`


OK, this is why (it is adding the groupings to the select menu):
<img width="516" alt="image" src="https://github.com/backdrop/backdrop-issues/assets/2423362/c1a58dbe-fc43-4273-8779-94a91534c48b">

Again, it shouldn't be the machine name of the module - it should be the human-readable name of the module instead.

**Comment by kiamlaluno** [comment]
Yes, `user_filters()` should be fixed for the contributed modules that uses it, as for now it is only marked deprecated. (I am going to add that detail to the IS.)

**Comment by kiamlaluno** [comment]
Thinking better, these issues should be keep separated. This issue should only mark the functions as deprecated, while the other issue needs to fix `user_filters()`, for which two different fixes are necessary.

**Comment by klonos** [comment]
Noting that in Backdrop, `user_filters()` is only called in one place: by `user_build_filter_query()`, which in turn is also only called in one place: `database_test_theme_tablesort()`. So, unlike in D7, not much to remove I guess.

The PR for addressing all these issues is very small and mostly adds docblock comments and `@todo`s (we cannot really add `@deprecated`, since that needs a reference to an actual Backdrop version - it will need to be done in 2.0). Still think separate issues/PRs @kiamlaluno?

**Comment by kiamlaluno** [comment]
> since that needs a reference to an actual Backdrop version

Backdrop core already uses `@deprecated for its own functions, for example [`backdrop_json_format()`](https://docs.backdropcms.org/api/backdrop/core%21includes%21common.inc/function/backdrop_json_format/1). (To get a full list, see the output of `git grep -n --fixed-strings '@deprecated`.)

In this case, `@deprecated since 1.28.1` or `@deprecated since 1.28.2` should be used, depending on which release will be created after the PR for this issue is merged.

**Comment by kiamlaluno** [comment]
The changes that should be done for this issue are described in the issue summary.

> `database_test_theme_tablesort()` should be rewritten to avoid `user_build_filter_query()`. `user_filters()` and `user_build_filter_query()` should be marked deprecated.

**Comment by klonos** [comment]
Fair enough @kiamlaluno 👍🏼 

Adding here some text from the original changes in the PR for #6456, to save us some time (we can tweak instead of create from scratch):

For the docblock of `user_filters()`:
```
 * Provided filters that could be applied to the user account listing.
 *
 * Same as with user_build_filter_query(), this function was used in the user
 * account listing page in Drupal 7. Since that listing is provided by Views in
 * Backdrop, this function here is slated for removal in Backdrop 2.0. Kept only
 * for backwards-compatibility purposes in Backdrop 1.0.
```

For the docblock of `user_build_filter_query()`:
```
 * Extended a query object for user account administration filters based on
 * session.
 *
 * Same as with user_filters(), this function was used in the user account
 * listing page in Drupal 7. Since that listing is provided by Views in
 * Backdrop, this function here is slated for removal in Backdrop 2.0. Kept only
 * for backwards-compatibility purposes in Backdrop 1.0.
```
